cmd/sink/filters/filters.go

```diff
@@ -167,27 +181,39 @@ func (f *Filters) createFilters(
 		} else {
 			deleteMap[namespaceGvk] = *deleteExpr
 		}
+		archiveOnDeleteExpr, err := ocel.CreateCelExprOr(resource.ArchiveOnDelete, globalArchiveOnDelete[gvk.String()])
+		if err != nil {
+			errList = append(errList, err)
+		} else {
+			archiveOnDeleteMap[namespaceGvk] = *archiveOnDeleteExpr
+		}
 	}
-	return archiveMap, deleteMap, errors.Join(errList...)
+	return archiveMap, deleteMap, archiveOnDeleteMap, errors.Join(errList...)
 }
 
 // updateNamespaceFilters updates the cel filters for namespace, from file. It differs from createNamespaceFilters by
 // deleting all filters for namespace before inserting the new ones.
-func (f *Filters) updateNamespaceFilters(namespace, file string, globalArchive, globalDelete map[string]string) error {
-	namespaceArchive, namespaceDelete, err := f.createFilters(namespace, file, globalArchive, globalDelete)
+func (f *Filters) updateNamespaceFilters(
+	namespace, file string,
+	globalArchive, globalDelete, globalArchiveOnDelete map[string]string,
+) error {
+	namespaceArchive, namespaceDelete, namespaceArchiveOnDelete, err := f.createFilters(namespace, file, globalArchive, globalDelete, globalArchiveOnDelete)
```

Comment:
Shouldn't you be checking that err returned by `createFilters` is empty before updating the filters? The code below uses the logic of deleting and recreating the filters based on what is returned by  `createFilters`, if `createFilters` returns a non-empty err all the existing filters will be removed.

Reply:
no, the goal of the filter update is to change the filter to match the new filters that the user provided, even if there were errors. the error from `createFilters` is returned to the caller of `updateNamespaceFilters` and will continue up the call stack until it gets printed to the container log.